### ISSUE TITLE
Support `@example(...).via("string description")` to track example provenance

### ISSUE THREAD

**Issue by Zac-HD** [HypothesisWorks/hypothesis#3506]
Over in https://github.com/Zac-HD/hypofuzz/issues/13, I mention using magic comments to track which `@example()` cases were automatically added by the fuzzer, to enably fully-automated maintanence of a static covering set.

It occured to me that provenance tracking might be a more generally desirable feature, and that a simple method which returns `self` might be a better way to implement this than comments.  Once I hit on the short-and-sweet name `via`, I was sold.  Probably requires some use of `typing.Protocol` to help type-checking tools understand that this is fine, with corresponding compatibility workaround on Python 3.7 (which predates `Protocol`).  The resulting code might look like:

```python
@example(0, 1)  # manually added
@example(1, 2).via("you can put anything you like in here")  # manually added
@example(3, 4).via("hypofuzz")  # added *and removable* by hypofuzz
@given(...)
def test(x: int, y: int):
    pass
```

**Comment by Zac-HD** [comment]
Guess what?  This is a *syntax error* before Python 3.9, because it requires [PEP-614](https://peps.python.org/pep-0614/).

I find the idea of a `# via: your string here` comment *aesthetically offensive*, but since we'll be supporting 3.8 until the EOL in October 2025 I guess I'll have to suck it up and deal 😬 

(plus we'll need to update the base Python env if we want to show `.via()` in docs without breaking anything, etc...)

**Comment by Zac-HD** [comment]
Closed by #3516.
